scout/adapter/mongo/clinvar.py

```diff
@@ -393,14 +393,14 @@ def delete_clinvar_object(self, object_id: str, object_type: str, submission_id:
             self.clinvar_collection.delete_one({"_id": object_id})
 
         # in any case remove reference to it in the submission object 'case_data' list field
-        self.clinvar_submission_collection.find_one_and_update(
-            {"_id": ObjectId(submission_id)}, {"$pull": {"case_data": object_id}}
-        )
 
         return self.clinvar_submission_collection.find_one_and_update(
-            {"_id": submission_id},
-            {"$set": {"updated_at": datetime.now()}},
-            return_document=pymongo.ReturnDocument.AFTER,
+            {"_id": ObjectId(submission_id)},
+            {
+                "$set": {"updated_at": datetime.now()},
+                "$pull": {"case_data": {"$regex": f"^{re.escape(object_id)}"}},
```

Comment:
```suggestion
                "$pull": {"case_data": {"$regex": f"{re.escape(object_id)}"}},
```

Reply:
Why? I think it is correct to match against something that starts with that string

Reply:
That was based off the variable name! 😁